pkg/addonmanager/addonclients/upgrader.go

```diff
@@ -35,6 +42,9 @@ func (f *FluxAddonClient) Upgrade(ctx context.Context, managementCluster *types.
 	if err := f.upgradeFilesAndCommit(ctx, newSpec); err != nil {
 		return nil, fmt.Errorf("failed upgrading Flux from bundles %d to bundles %d: %v", currentSpec.Bundles.Spec.Number, newSpec.Bundles.Spec.Number, err)
 	}
+	if err := f.flux.DeleteFluxSystemSecret(ctx, managementCluster, newSpec.GitOpsConfig.Spec.Flux.Github.FluxSystemNamespace); err != nil {
```

Comment:
Would flux stop working until the secret is recreated? I wonder if we should pause the flux reconcile before this and enable just after finishing the the flux component upgrade

Reply:
no, the `flux bootstrap` cmd thats executed right after will recreate the secrets.